litestar/_signature/models/msgspec_signature_model.py

```diff
@@ -0,0 +1,132 @@
+
+    def to_dict(self) -> dict[str, Any]:
+        """Normalize access to the signature model's dictionary method, because different backends use different methods
+        for this.
+
+        Returns: A dictionary of string keyed values.
+        """
+        return asdict(self)
+
+    @classmethod
+    def populate_signature_fields(cls) -> None:
+        ...
+
+    @classmethod
+    def create(
+        cls,
+        fn_name: str,
+        fn_module: str | None,
+        parsed_signature: ParsedSignature,
+        dependency_names: set[str],
+        type_overrides: dict[str, Any],
+    ) -> type[SignatureModel]:
+        struct_fields: list[tuple[str, Any, Any]] = []
+        signature_fields: dict[str, SignatureField] = {}
+
+        for parameter in parsed_signature.parameters.values():
+            annotation = type_overrides.get(parameter.name, parameter.parsed_type.annotation)
+
+            field_extra: dict[str, Any] = {"parsed_parameter": parameter}
+            meta_kwargs: dict[str, Any] = {}
+
+            if kwargs_container := parameter.kwarg_container:
+                field_extra["kwargs_model"] = kwargs_container
+                if isinstance(kwargs_container, DependencyKwarg):
+                    annotation = annotation if not kwargs_container.skip_validation else Any
+                    default = kwargs_container.default if kwargs_container.default is not Empty else NODEFAULT
+                else:
+                    param_dict = simple_asdict(kwargs_container)
+                    field_extra.update(param_dict)
+                    meta_kwargs = {
+                        k: v
+                        for k in (
+                            "gt",
+                            "ge",
+                            "lt",
+                            "le",
+                            "multiple_of",
+                            "pattern",
+                            "min_length",
+                            "max_length",
+                        )
+                        if (v := getattr(kwargs_container, k))
+                    }
+                    default = NODEFAULT
+            else:
+                default = parameter.default if parameter.has_default else NODEFAULT
+
+            struct_fields.append(
+                (parameter.name, Annotated[annotation, Meta(extra=field_extra, **meta_kwargs)], default)
+            )
+            signature_fields[parameter.name] = SignatureField.create(
+                field_type=annotation,
+                name=parameter.name,
+                default_value=Empty if default is NODEFAULT else default,
+                kwarg_model=kwargs_container,
+                extra=field_extra,
+            )
+
+        return defstruct(  # type:ignore[return-value]
+            f"{fn_name}_signature_model",
+            struct_fields,
+            bases=(MsgspecSignatureModel,),
+            module=fn_module or "",
```

Comment:
This workaround can be removed with the next msgspec release (https://github.com/jcrist/msgspec/issues/444)

Reply:
Nice!